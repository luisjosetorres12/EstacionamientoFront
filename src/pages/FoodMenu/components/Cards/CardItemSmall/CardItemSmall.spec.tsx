/* eslint-disable react/jsx-props-no-spreading, @typescript-eslint/no-var-requires, global-require */
import React from 'react';
import {
  CardItemSmall,
  CardItemSmallProps,
} from 'pages/FoodMenu/components/Cards/CardItemSmall/index';
import { render } from '@testing-library/react';

jest.mock('react-intersection-observer', () => {
  const { factory } = require('tests/mocks/react-intersection-observer.mock');
  return factory();
});

describe('CardItemSmall Test', () => {
  let props: CardItemSmallProps;

  beforeEach(() => {
    props = {
      coverImg: 'food.jpg',
      description: 'some description',
      descriptionDirection: 'row',
      onImageClick: jest.fn(),
      title: 'some title',
    };
  });

  it('should match snapshot', () => {
    const { container } = render(<CardItemSmall {...props} />);
    expect(container).toMatchSnapshot();
  });
});
