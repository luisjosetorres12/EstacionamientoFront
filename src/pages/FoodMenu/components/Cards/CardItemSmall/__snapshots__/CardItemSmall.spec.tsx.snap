// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardItemSmall Test should match snapshot 1`] = `
<div>
  <article
    class="makeStyles-paper-2"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        class="MuiGrid-root makeStyles-item-3 MuiGrid-item MuiGrid-grid-xs-8"
      >
        <div
          class="MuiGrid-root MuiGrid-item"
        >
          <button
            class="MuiButtonBase-root"
            tabindex="0"
            type="button"
          >
            <img
              alt="Food item"
              class="makeStyles-img-1"
              src="/images/food.jpg"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiGrid-root makeStyles-description-4 makeStyles-description-7 MuiGrid-item"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle1"
          >
            some title
          </h6>
          <h6
            class="MuiTypography-root makeStyles-descriptionInner-5 makeStyles-descriptionInner-8 MuiTypography-subtitle1"
          >
            some description
          </h6>
        </div>
      </div>
      <div
        class="MuiGrid-root makeStyles-rightContainer-6 MuiGrid-item MuiGrid-grid-xs-4"
      />
    </div>
  </article>
</div>
`;
